TUTORIAL.md

```diff
@@ -0,0 +1,208 @@
+```
+
+###Step 3: Set up our form
+The next piece of this puzzle is our form. We have our attribute writer set up, so now let's add an input to our form so we can actually add a new tag.  
+
+####`posts/_form.html.erb`
+
+Notice we are using `#fields_for`. This method creates a scope around a specific model object, which is `#form_for`, but doesn't create the form tags themselves. This makes `#fields_for` suitable for specifying additional model objects in the same form. Reade more <a href="http://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-fields_for">here</a>.
+
+If you inspect the `html` of our form, you can see that our params are set up nicely, thanks to `#accepts_nested_attributes_for`, which we have included in our `Post` model.
+
+`<input type="text" name="post[tags_attributes][0][name]" id="post_tags_attributes_0_name">`
+
+```ruby
+<div class="field">
+  <%= f.fields_for(:tags) do |tag_form| %>
+    <%= tag_form.label :name %>
+    <%= tag_form.text_field :name %>
+  <% end %>
+</div>
+```
+
+If you delete `accepts_nested_attributes_for :tags` from your `Post` model and refresh your browser, you will see the params have changed to the following.
+
+`<input type="text" name="post[tags][name]" id="post_tags_name">`
+
+####Completed Form
+
+```ruby
+<%= form_for(@post) do |f| %>
+  <% if @post.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@post.errors.count, "error") %> prohibited this post from being saved:</h2>
+      <ul>
+      <% @post.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= f.label :name %><br>
+    <%= f.text_field :name %><br>
+    <%= f.label :content %><br>
+    <%= f.text_area :content %><br>
+    Tags:
+  </div>
+
+  <div class="field">
+    <%= f.fields_for(:tags) do |tag_form| %>
+      <%= tag_form.label :name %>
+      <%= tag_form.text_field :name %>
+    <% end %>
+  </div>
+
+  <div class="field">
+    <%= f.collection_check_boxes :tag_ids, Tag.all, :id, :name %><br>
+  </div>
+
+  <div class="actions">
+    <%= f.submit %>
+  </div>
+
+<% end %>
+```
+
+###Step 3: Set up your PostsContrller
+
+### PostsController
+
+Ok so now we have our model set up and our form in place, let's visit `localhost:3000/posts/new` and see what our updated form looks like. Hmmm, we were expecting a text field so that we can add our new tag, which is what this code should be creating.
+
+```
+<div class="field">
+  <%= f.fields_for(:tags) do |tag_form| %>
+    <%= tag_form.label :name %>
+    <%= tag_form.text_field :name %>
+  <% end %>
+</div>
+```
+<img src="https://github.com/learn-co-curriculum/rails-blog-nested-forms/blob/ea837ba87a44f7cb39e8d10233dcd68984f9b40a/app/assets/images/nobuild.jpg?raw=true" width="50%">
+
+So what is going on? There is one more thing we need to do for our form to render properly, which is update the `new` action of our `Posts` controller.
```

Comment:
maybe also play around with happens if you put a name in the Tag that you create? how does the form respond?